sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -97,6 +98,12 @@ class GpuHashAggregateMeta(
       aggregateAttributes ++
       resultExpressions
 
+  def isSupported(t: DataType) = t match {
```

Comment:
Why do we need to change the aggregate operators?  Did you run any tests that showed that maps would be able to sneak in to them?  The generic type checking should already disallow them.

Reply:
You are right. I was being too defensive and missed that. I will take it out of the operators and keep it in this hashAgg meta.
I am writing up a test to make sure we didn't really break anything - to answer your question on whether this really truly works the way we want since like I said my target test query doesn't really make the cut for full fledged testing of this change.

Reply:
Ok so this is what is throwing me off even though it is doing the right thing after all.
If I keep the change in Hash agg meta and run : 
```
  IGNORE_ORDER_ALLOW_NON_GPU_testSparkResultsAreEqual(
    "test hash agg with maps",
    frameFromParquet("map_of_strings.snappy.parquet"),
    execsAllowedNonGpu = Seq("HashAggregateExec", "AggregateExpression", "AttributeReference",
      "Alias", "Count", "FileSourceScanExec"),
    conf = new SparkConf().set(RapidsConf.EXPLAIN.key, "ALL")) {
    frame => {
      val res = frame.groupBy("id").agg(count("mapField"))
      res.explain
      res
    }
  }
```
where mapField is maptype, the hash aggregate falls back but `Count` which is on this maptype does not _say_ that it does.
```
!Exec <HashAggregateExec> cannot run on GPU because not all expressions can be replaced
  *Expression <AggregateExpression> count(mapField#295) could run on GPU
    *Expression <Count> count(mapField#295) could run on GPU
  *Expression <Alias> count(mapField#295)#300L AS count(mapField)#302L could run on GPU
  !Exec <ShuffleExchangeExec> cannot run on GPU because Columnar exchange without columnar children is inefficient
    *Partitioning <HashPartitioning> could run on GPU
    !Exec <HashAggregateExec> cannot run on GPU because not all expressions can be replaced
      *Expression <AggregateExpression> partial_count(mapField#295) could run on GPU
        *Expression <Count> count(mapField#295) could run on GPU
      !Exec <ShuffleExchangeExec> cannot run on GPU because unsupported data types in output: MapType(StringType,StringType,true); Columnar exchange without columnar children is inefficient; unsupported data types in input: MapType(StringType,StringType,true)
        *Partitioning <RoundRobinPartitioning> could run on GPU
        !Exec <FileSourceScanExec> cannot run on GPU because GpuParquetScan does not support fields of type MapType(StringType,StringType,true); unsupported data types in output: MapType(StringType,StringType,true)
```

like here : _Expression <Count> count(mapField#295) could run on GPU_
I will have to debug more on why unless I am missing something obvious. At least to me it should say count cannot run on the gpu due to unsupported type.

Reply:
Debugged some more and turns out it looks at the o/p data type not the incoming data type. The attribute is called out as not supported but not the operator.
Debug log.
```
KUHU expr=mapField#295 issupported all=false expr.dataType=MapType(StringType,StringType,true)
KUHU expr=count(mapField#295) issupported all=true expr.dataType=LongType
```

Reply:
It is just that the exprs like `mapField` that are unsupported dont reflect in the outer expression explain. which is ok I guess. @revans2 I have pushed a change so that only HashAggregateMeta whitelists this MapType. Let me know if I missing something here. We could fix how the explain really looks if u think that is an issue as a separate PR.